### PR TITLE
Add support for mobile suggest pings

### PR DESCRIPTION
**Pull request by cbguder**
Adds support for mobile suggest pings in ParseReportingUrl

Also:
* Refactors ParseReportingUrl quite heavily (reviewing each diff individually might help)
* Adds a new metric for low IP reputation reports
* Adds CODEOWNERS for Ads Eng (Ads Eng will need to be granted write access to the repo by repo owners)

### CODE REVIEW

**Comment by quiiver** [comment]
@whd What do we need to get @mozilla/ads-engineering write access to the repository?

File: ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/ParseReportingUrl.java [hunk 1]
```diff
@@ -295,6 +295,7 @@ private static void addAdditionalDimensionsForTopSitesClicks(BuildReportingUrl b
     }
 
     if (ipReputation != null && ipReputation < IP_REPUTATION_THRESHOLD) {
+      PerDocTypeCounter.inc(attributes, "rejected_low_ip_reputation");
```
Comment (quiiver) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/ParseReportingUrl.java:313]: Its good to have this metric here but i'm fairly certain we have turned off ip reputation checking as it was expensive/slow and not providing much value.

**Comment by whd** [comment]
> @whd What do we need to get @mozilla/ads-engineering write access to the repository?

I don't think there are any specific considerations here, so any repo admin can add that team to collaborators. I'd recommend adding `CODEOWNERS` with either `@mozilla/data-platform-infra-wg` or `@mozilla/telemetry` as the default if you're adding more teams with write access.

File: ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/ParseReportingUrl.java (hunk 1 above)
Comment (quiiver) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/ParseReportingUrl.java:313]: I say leave it in to verify.

**Review by quiiver — APPROVED** [review]
I'm going to go ahead and approve but please make sure to test these changes in the dev environment as described in the documentation here: https://mozilla.github.io/gcp-ingestion/ingestion-beam/contextual-services-job/#test-deployment

**Pull request merged**